packages/itwinui-react/src/core/Checkbox/Checkbox.tsx

```diff
@@ -116,13 +130,20 @@ export const Checkbox = React.forwardRef((props, ref) => {
           [`iui-${status}`]: !!status,
           'iui-loading': isLoading,
         },
-        className,
+        wrapperClassName ? wrapperClassName : className,
       )}
-      style={style}
+      {...restWrapperProps}
     >
       {checkbox}
       {label && (
-        <Box as='span' className='iui-checkbox-label'>
+        <Box
+          as='span'
+          className={
+            (cx('iui-checkbox-label'),
+            labelClassName ? labelClassName : className)
```

Comment:
With this code, if user has labeled checkbox and adds `custom-classname`, but not labelClassName/wrapperClassName, html structure would be this:
![image](https://github.com/iTwin/iTwinUI/assets/83585998/3ea4ef9a-935a-4c33-871e-67bba66ed912)

Reply:
I wrote it like that so that the `label` and `wrapper` would share a className with the parent component by default, unless the user goes out of their way to specify custom classNames for `label` and `wrapper`. Should their classNames both be different by default?

Reply:
Why they need to share className?

Reply:
They don't have to, I just figured some users might want them to. If they shouldn't share classNames by default, then I'll change the code so they don't

Reply:
If user wants to add styles to certain part of component, they need to use props for that part of component.
Remove this check.

Reply:
Removed the checks.

Two unit tests are now failing again. I think it's because they're assuming that setting the `className` variable for the component would still assign it to both the `input` and the `label`, when this isn't the case anymore. I could rewrite the tests so that they make sure the label *isn't* automatically assigned the same `className` as the `input` when only the `input` is given the assignment. 
![image](https://github.com/iTwin/iTwinUI/assets/44583237/9254ebf4-990a-4518-a4bf-76c9d34193a0)
![image](https://github.com/iTwin/iTwinUI/assets/44583237/3847a55c-7ee0-4c85-9bd2-5e1540bedf52)

Reply:
i think those two tests can be adjusted/removed because the behavior that they were testing has been changed significantly.

Reply:
Modified` "Should isomorphically apply class on %s"` to only test `input`, and also removed the final failing test.

Reply:
it is no longer isomorphic so it's not testing anything special and can be removed

imo you should have a single unit test that tests adding separate custom classes all three elements (`className`, `labelProps`, `wrapperProps`)

Reply:
Done

Reply:
you're only testing two cases in there, missing a test for `className` which gets applied on the input.

Reply:
Added